src/Common/Caching/CacheClient.cs

```diff
@@ -394,10 +407,44 @@ protected void CreateParentDirectory(string filePath)
             return (null, null);
         }
 
-        Func<CancellationToken, Task> placeFilesAsync = (ct) => cacheEntry.PlaceFilesAsync(
-            context,
-            nodeBuildResult.Outputs.ToDictionary(o => Path.Combine(RepoRoot, o.Key), o => o.Value),
-            ct);
+        Func<CancellationToken, Task> placeFilesAsync = async (ct) =>
+        {
+            List<Task> tasks = new(nodeBuildResult.PackageFilesToCopy.Count + 1);
+
+            Dictionary<string, ContentHash> outputsToPlace = new(nodeBuildResult.Outputs.Count - nodeBuildResult.PackageFilesToCopy.Count);
+            foreach (KeyValuePair<string, ContentHash> kvp in nodeBuildResult.Outputs)
+            {
+                string destinationAbsolutePath = Path.Combine(RepoRoot, kvp.Key);
+                if (nodeBuildResult.PackageFilesToCopy.TryGetValue(kvp.Key, out string? packageFile))
+                {
+                    tasks.Add(Task.Run(
+                        () =>
+                        {
+                            string sourceAbsolutePath = Path.Combine(_nugetPackageRoot, packageFile);
+                            CreateParentDirectory(destinationAbsolutePath);
+
+                            if (_canCloneInNugetCachePath && AreOnSameDrive(sourceAbsolutePath, destinationAbsolutePath))
```

Comment:
Can we do this check at startup and warn if 1) repository is on volume with CoW *and* 2) nuget folder is not?

Reply:
Probably same with the cache actually

Reply:
![image](https://github.com/microsoft/MSBuildCache/assets/6445614/2d640326-d291-450e-96a0-d95ef8f242f3)